### ISSUE TITLE
Text magnification feature for the Quill Editor

### ISSUE THREAD

**Issue by AdiSuresh** [singerdmx/flutter-quill#1504]
I'm making a note-taking app using the `flutter_quill` package. While using the editor, I noticed that dragging the cursor cannot magnify the text.

Since I’m aware that the `TextMagnifierConfiguration` class is available for the `TextField` widget and text magnification is enabled by default, I thought there might be something similar in the `QuillEditorConfigurations` constructor but could not find any.

Is this feature already available or is there a workaround to make it work? It’d be more convenient for users to edit their notes by moving the cursor using the magnifier.

**Comment by singerdmx** [comment]
Yes it should be a nice feature

**Comment by molihuan** [comment]
> I'm making a note taking app using the flutter_quill package. While using the editor, I noticed that the text cannot be magnified by dragging the cursor.
> 
> Since I’m aware that the TextMagnifierConfiguration class is available for the TextField widget and text magnification is enabled by default, I thought there might be something similar in the QuillEditorConfigurations constructor but could not find any.
> 
> Is this feature already available or is there a workaround to make it work? I think it’d be more convenient for users to edit their notes by moving the cursor easily using the magnifier.
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 

I am also working on an open source note-taking app, interested in discussing related issues or sharing experiences together?

**Comment by freshtechtips** [comment]
> > I'm making a note taking app using the flutter_quill package. While using the editor, I noticed that the text cannot be magnified by dragging the cursor.
> > 
> > Since I’m aware that the TextMagnifierConfiguration class is available for the TextField widget and text magnification is enabled by default, I thought there might be something similar in the QuillEditorConfigurations constructor but could not find any.
> > 
> > Is this feature already available or is there a workaround to make it work? I think it’d be more convenient for users to edit their notes by moving the cursor easily using the magnifier.
> > 
> > 
> > 
> > 
> > 
> > 
> > 
> > 
> > 
> > 
> > 
> 
> I am also working on an open source note-taking app, interested in discussing related issues or sharing experiences together?

Sure, why not.

**Comment by freshtechtips** [comment]
> I'm making a note taking app using the flutter_quill package. While using the editor, I noticed that the text cannot be magnified by dragging the cursor.
> 
> Since I’m aware that the TextMagnifierConfiguration class is available for the TextField widget and text magnification is enabled by default, I thought there might be something similar in the QuillEditorConfigurations constructor but could not find any.
> 
> Is this feature already available or is there a workaround to make it work? I think it’d be more convenient for users to edit their notes by moving the cursor easily using the magnifier.
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 

If you have in mind any ideas about libraries or apps that implemented this feature 

Feel free to share the presentation using images or videos

**Comment by freshtechtips** [comment]
> I'm making a note taking app using the flutter_quill package. While using the editor, I noticed that the text cannot be magnified by dragging the cursor.
> 
> Since I’m aware that the TextMagnifierConfiguration class is available for the TextField widget and text magnification is enabled by default, I thought there might be something similar in the QuillEditorConfigurations constructor but could not find any.
> 
> Is this feature already available or is there a workaround to make it work? I think it’d be more convenient for users to edit their notes by moving the cursor easily using the magnifier.
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 

If you have in mind any ideas about libraries or apps that implemented this feature 

Feel free to share the presentation using images or videos

**Comment by singerdmx** [comment]
This is probably built-in function in TextField .
Just need right way to invoke it.

**Comment by singerdmx** [comment]
https://api.flutter.dev/flutter/material/TextField/magnifierConfiguration.html

**Comment by singerdmx** [comment]
I assume this is only for mobile?

**Comment by AdiSuresh** [comment]
> > I'm making a note taking app using the flutter_quill package. While using the editor, I noticed that the text cannot be magnified by dragging the cursor.
> > Since I’m aware that the TextMagnifierConfiguration class is available for the TextField widget and text magnification is enabled by default, I thought there might be something similar in the QuillEditorConfigurations constructor but could not find any.
> > Is this feature already available or is there a workaround to make it work? I think it’d be more convenient for users to edit their notes by moving the cursor easily using the magnifier.
> 
> I am also working on an open source note-taking app, interested in discussing related issues or sharing experiences together?

Sure. I don’t mind having a look at your repo, but only if I get some time...

**Comment by AdiSuresh** [comment]
> > I'm making a note taking app using the flutter_quill package. While using the editor, I noticed that the text cannot be magnified by dragging the cursor.
> > Since I’m aware that the TextMagnifierConfiguration class is available for the TextField widget and text magnification is enabled by default, I thought there might be something similar in the QuillEditorConfigurations constructor but could not find any.
> > Is this feature already available or is there a workaround to make it work? I think it’d be more convenient for users to edit their notes by moving the cursor easily using the magnifier.
> 
> If you have in mind any ideas about libraries or apps that implemented this feature
> 
> Feel free to share the presentation using images or videos

I’m attaching a video for reference. Many android apps either have the magnifier or the drag cursor widget (I don’t know what it’s actually called). I use Google Docs quite often and it has both:

https://github.com/singerdmx/flutter-quill/assets/50865181/25491c0a-5cc9-4edb-a518-12baae7559e1

**Comment by AdiSuresh** [comment]
> I assume this is only for mobile?

Yeah it seems like it. Personally I think that all touchscreen enabled devices must have this feature.

**Comment by molihuan** [comment]
> > > I'm making a note taking app using the flutter_quill package. While using the editor, I noticed that the text cannot be magnified by dragging the cursor.
> > > Since I’m aware that the TextMagnifierConfiguration class is available for the TextField widget and text magnification is enabled by default, I thought there might be something similar in the QuillEditorConfigurations constructor but could not find any.
> > > Is this feature already available or is there a workaround to make it work? I think it’d be more convenient for users to edit their notes by moving the cursor easily using the magnifier.
> > 
> > I am also working on an open source note-taking app, interested in discussing related issues or sharing experiences together?
> 
> Sure. I don’t mind having a look at your repo, but only if I get some time...

It's still on the drawing board.
link:https://github.com/molihuan/HLVideoNote
But it's Chinese. It hasn't been translated into English yet。

**Comment by freshtechtips** [comment]
> > > I'm making a note taking app using the flutter_quill package. While using the editor, I noticed that the text cannot be magnified by dragging the cursor.
> > > Since I’m aware that the TextMagnifierConfiguration class is available for the TextField widget and text magnification is enabled by default, I thought there might be something similar in the QuillEditorConfigurations constructor but could not find any.
> > > Is this feature already available or is there a workaround to make it work? I think it’d be more convenient for users to edit their notes by moving the cursor easily using the magnifier.
> > 
> > 
> > If you have in mind any ideas about libraries or apps that implemented this feature
> > Feel free to share the presentation using images or videos
> 
> I’m attaching a video for reference. Many android apps either have the magnifier or the drag cursor widget (I don’t know what it’s actually called). I use Google Docs quite often and it has both:
> 
>  Screenrecorder-2023-11-07-22-47-17-176.mp4

Thank you for sharing that. We will work on it soon.

**Comment by freshtechtips** [comment]
> This is probably built-in function in TextField . Just need right way to invoke it.

![image](https://github.com/singerdmx/flutter-quill/assets/73608287/0a0a7dc9-d899-4c90-baa1-9cb6851dd2fd)

Probably in `QuilRawEditorMultiChildRenderObjectWidget`

**Comment by freshtechtips** [comment]
> https://api.flutter.dev/flutter/material/TextField/magnifierConfiguration.html

The `TextMagnifierConfiguration` has `MagnifierBuilder` which returns a widget

```dart
typedef MagnifierBuilder = Widget? Function(
    BuildContext context,
    MagnifierController controller,
    ValueNotifier<MagnifierInfo> magnifierInfo,
);
```

**Comment by freshtechtips** [comment]
I just added the `TextMagniferConfiguration` in the `QuillEditorConfigurations`

in this [commit](https://github.com/singerdmx/flutter-quill/commit/614bf1a526a122a848c67f747b85140be7bbb2b7)

We are not using it and it has no effect right now but don't worry I marked it as `@experimental`
and added a comment as well to make it clear

it's a very first step but I'm sure we will find a solution

**Comment by xalanq** [comment]
any update?

**Comment by EchoEllet** [comment]
Reopened since it's reverted in #2406, we don't have plans on adding this feature back since it causes many issues, and we **strongly prefer stability over adding new buggy features that are not well-executed**.
